src/Common/src/TypeSystem/Ecma/EcmaMethod.cs

```diff
@@ -428,7 +428,15 @@ public override PInvokeMetadata GetPInvokeMethodMetadata()
             Debug.Assert((int)MethodImportAttributes.CharSetUnicode == (int)PInvokeAttributes.CharSetUnicode);
             Debug.Assert((int)MethodImportAttributes.SetLastError == (int)PInvokeAttributes.SetLastError);
 
-            return new PInvokeMetadata(moduleName, name, (PInvokeAttributes)import.Attributes);
+            DllImportSearchPath dllImportSearchPath = this.GetDllImportSearchPath();
+
+            // if DefaultDllImportSearchPathAttribute is not assigned on the method
+            // check to see whether it is assigned on the containing assembly
+            if (dllImportSearchPath == DllImportSearchPath.None)
+            {
+                dllImportSearchPath = ((EcmaAssembly)Module).GetDllImportSearchPath();
```

Comment:
I usually put an assert in places like this:

```csharp
Debug.Assert(Module is IAssemblyDesc, "Multi-module assemblies");
```

If we ever add support for multi-module assemblies, it's a nice breadcrumb to search for.